sklearn/random_projection.py

```diff
@@ -537,16 +499,34 @@ def _make_random_matrix(self, n_components, n_features):
 
         Returns
         -------
-        components : {ndarray, sparse matrix} of shape \
-                (n_components, n_features)
-            The generated random matrix. Sparse matrix will be of CSR format.
-
+        components : ndarray of shape (n_components, n_features)
+            The generated random matrix.
         """
         random_state = check_random_state(self.random_state)
         return _gaussian_random_matrix(
             n_components, n_features, random_state=random_state
         )
 
+    def transform(self, X):
+        """Project the data by using matrix product with the random matrix.
+
+        Parameters
+        ----------
+        X : {ndarray, sparse matrix} of shape (n_samples, n_features)
+            The input data to project into a smaller dimensional space.
+
+        Returns
+        -------
+        X_new : ndarray of shape (n_samples, n_components)
+            Projected array.
+        """
+        check_is_fitted(self)
+        X = self._validate_data(
+            X, accept_sparse=["csr", "csc"], reset=False, dtype=[np.float64, np.float32]
+        )
+
+        return X @ self.components_.T
```

Comment:
I'm surprised this works when `X` is a sparse matrix and `components_` is a ndaray.

Reply:
```py
import numpy as np
from scipy.sparse import csr_matrix

X = np.random.RandomState(0).random_sample((100, 100))
X_csr = csr_matrix(X)

np.max(np.abs(X @ X_csr -  X @ X))
# 2.4868995751603507e-14
```

Reply:
I always need to run a similar experiment to remember what happens with `sparse @ dense`, like the return is dense.

Anyways, this is a non-issue so I'm resolving the conversation.